### PR TITLE
[clang][driver] Set TLSDESC as the default for Android on RISC-V

### PR DESCRIPTION
**Pull request by ilovepi**

### CODE REVIEW

**Comment by pirama-arumuga-nainar** [comment]
Can you add a test similar to clang/test/Driver/emulated-tls.cpp?

**Comment by ilovepi** [comment]
> Can you add a test similar to clang/test/Driver/emulated-tls.cpp?

I knew I was forgetting something when I threw this up. Thank you!

**Review by pirama-arumuga-nainar — APPROVED** [review]

**Comment by enh-google** [comment]
i think my suggestion would be instead of
```
    return isAndroid() && (!isAndroidVersionLT(29) || isRISCV64());
```
just go with
```
    return isAndroid() && isRISCV64();
```

that solves today's problem, and we can worry about arm64 later[1]. (and x86-64 if/when that's implemented. and 32-bit never.)

____
1. why, when arm64 tlsdesc already shipped, and rv64 hasn't shipped yet? for _exactly_ that reason :-) with rv64 i don't even have to [have rprichard] think about app compat issues, say, or [have danalbert] think about communicating the change to ndk users, etc...

**Comment by ilovepi** [comment]
Sounds good. I’ll update this when I have a chance tomorrow.

File: clang/test/Driver/tls-dialect.c [hunk 1]
```diff
@@ -3,6 +3,11 @@
 // RUN: %clang -### --target=riscv64-linux %s 2>&1 | FileCheck --check-prefix=NODESC %s
 // RUN: %clang -### --target=x86_64-linux -mtls-dialect=gnu %s 2>&1 | FileCheck --check-prefix=NODESC %s
 
+/// Android supports TLSDESC by default after Android version 29 and all RISC-V
+/// TLSDESC is not on by default in Linux, even on RISC-V
+// RUN: %clang -### --target=riscv64-android %s 2>&1 | FileCheck --check-prefix=DESC %s
+// RUN: %clang -### --target=riscv64-linux %s 2>&1 | FileCheck --check-prefix=NODESC %s
```
Comment (MaskRay) [inline clang/test/Driver/tls-dialect.c:9]: Delete `riscv64-linux` check. It's redundant.

**Review by MaskRay — APPROVED** [review]
LGTM once the `!isAndroidVersionLT(29)` check is removed.

File: clang/test/Driver/tls-dialect.c (hunk 1 above)
Comment (ilovepi) [reply clang/test/Driver/tls-dialect.c]: That's a good point. I went ahead and updated the comment to point that out.

**Pull request merged**